x/ibc/02-client/keeper/keeper.go

```diff
@@ -148,10 +150,14 @@ func (k Keeper) GetSelfConsensusState(ctx sdk.Context, height uint64) (exported.
 // the iterator will close and stop.
 func (k Keeper) IterateClients(ctx sdk.Context, cb func(exported.ClientState) bool) {
 	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, ibctypes.KeyClientPrefix)
+	iterator := sdk.KVStorePrefixIterator(store, ibctypes.KeyClientStorePrefix)
 
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
+		keySplit := strings.Split(string(iterator.Key()), "/")
+		if keySplit[len(keySplit)-1] != "clientState" {
```

Comment:
I'm a bit lost, why does it skip the value if the last item of the key doesn't match the string?

Reply:
Because presumably consensus states are also under this prefix?

Reply:
Yup exactly, so are clientTypes

Reply:
Not a big deal here, so we follow ics spec. Really bad in the channel cases hence why we prefix them